references/optical_flow/train.py

```diff
@@ -115,21 +120,21 @@ def inner_loop(blob):
         inner_loop(blob)
         num_processed_samples += blob[0].shape[0]  # batch size
 
-    num_processed_samples = utils.reduce_across_processes(num_processed_samples)
-    print(
-        f"Batch-processed {num_processed_samples} / {len(val_dataset)} samples. "
-        "Going to process the remaining samples individually, if any."
-    )
-
-    if args.rank == 0:  # we only need to process the rest on a single worker
-        for i in range(num_processed_samples, len(val_dataset)):
-            inner_loop(val_dataset[i])
+    if args.distributed:
+        num_processed_samples = utils.reduce_across_processes(num_processed_samples)
+        print(
+            f"Batch-processed {num_processed_samples} / {len(val_dataset)} samples. "
+            "Going to process the remaining samples individually, if any."
+        )
+        if args.rank == 0:  # we only need to process the rest on a single worker
+            for i in range(num_processed_samples, len(val_dataset)):
+                inner_loop(val_dataset[i])
 
     logger.synchronize_between_processes()
```

Comment:
This should be part of the `if args.distributed:` block

```suggestion
        logger.synchronize_between_processes()
```